src/core/server/logging/appenders/rolling_file/strategies/numeric/pattern_matcher.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const escape = (string: string) => string.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&');
+
+const createNumericMatcher = (fileBaseName: string, pattern: string): RegExp => {
+  let extStart = fileBaseName.indexOf('.');
```

Comment:
Do you want `lastIndexOf` here instead? Would this incorrectly say that `foo.bar.log` has an extension of `bar.log`?

```suggestion
  let extStart = fileBaseName.lastIndexOf('.');
```

Reply:
That was intentional. `foo.bar.log` should (imho) roll to `foo-1.bar.log`? That's usually what is done for multi-extension I think?

Reply:
Ah, I think I've personally seen `foo.bar-1.log` in a past life, but I'm fine to defer to you if this was intentional. I was thinking if someone had `kibana.audit.json`, that `kibana-1.audit.json` would look a bit funny.

Reply:
I was think of `foo.tar.gz`, where the rename would make more sense to be `foo-1.tar.gz`. But I guess it depends then. We could eventually add an option later. For now, I good with either option.

Reply:
> For now, I good with either option.

Same. Feel free to leave it as-is